helix-view/src/editor.rs

```diff
@@ -238,9 +238,28 @@ impl Editor {
                     self.documents[view.doc].selection(view.id).clone(),
                 );
 
+                let doc = doc!(self);
+                // If the current view is an empty scratch buffer and is not displayed in any other views, delete it.
+                // Boolean value is determined before the creation of the `view` variable because the operation
+                // requires a borrow of `self.tree`, which is mutably borrowed when `view` is created.
+                let remove_empty_scratch = !doc.is_modified()
+                    // If the buffer has no path and is not modified, it is an empty scratch buffer.
+                    && doc.path().is_none()
+                    // Ensure the buffer is not displayed in any other splits.
+                    && !self
+                        .tree
+                        .traverse()
+                        .any(|(_, view)| view.doc == doc.id && view.id != view!(self).id);
```

Comment:
How about first changing the `view.doc` and `view.offset` (L263/264)? This way you don't need to check for `view.id != view!(self).id`) here and we can extract this traversal into a method

Reply:
That would require using `view_mut!` twice: once to set `view.doc` and `view.offset`, and once after borrowing `self.tree` for the check. We could also do this:
```rs
self.tree.traverse().filter(|(_, view)| view.doc == doc.id).count() > 1
```

Reply:
Ah I see. We want to at least get rid of the `view!` calls inside `any`, since that costs as much as a `view_mut` but would be called per iteration here. On `L235` we already borrow `view` so you should be able to reuse that:

- change L235 to `let (view, doc) = current_ref!(self)`
- remove L241 (`let doc = doc!(self);`)
- change any to `.any(|(_, v)| v.doc == doc.id && v.id != view.id);`